MmSupervisorPkg/Drivers/MmPeiLaunchers/MmDxeSupport.c

```diff
@@ -709,6 +712,276 @@ SmmIplSetVirtualAddressNotify (
+  //
+  Status = SupvCommunicationCommunicate (&mMmSupvCommunication, mCommunicateHeader, &Size);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a Failed to communicate to MM through supervisor channel - %r!!\n", __FUNCTION__, Status));
+    Status = EFI_DEVICE_ERROR;
+    goto Done;
+  }
+
+  // Re-check the return status on the new buffers.
+  gMmCorePrivate = (MM_CORE_PRIVATE_DATA *)NewMmCoreBuffer;
+  if ((UINTN)gMmCorePrivate->ReturnStatus != 0) {
+    Status = gMmCorePrivate->ReturnStatus;
+    DEBUG ((DEBUG_ERROR, "%a Failed to communicate to MM to switch core mailbox - %r!!\n", __FUNCTION__, Status));
+    Status = EFI_DEVICE_ERROR;
+    goto Done;
+  }
+
+  mCommunicateHeader = (EFI_SMM_COMMUNICATE_HEADER *)NewSupvCommBuffer;
+  RequestHeader      = (MM_SUPERVISOR_REQUEST_HEADER *)mCommunicateHeader->Data;
+  if ((UINTN)RequestHeader->Result != 0) {
+    Status = ENCODE_ERROR ((UINTN)RequestHeader->Result);
+    DEBUG ((DEBUG_ERROR, "%a Failed to switch communication channel - %r!!\n", __FUNCTION__, Status));
+    Status = EFI_DEVICE_ERROR;
+    goto Done;
+  }
+
+  // Populate the returned buffer
+  CopyMem (UpdatedCommBuffer, NewCommBuffer, sizeof (*UpdatedCommBuffer));
+
+  // Update the global variables
+  mMmUserCommonBuffer         = NewUserCommBuffer;
+  mMmSupvCommonBuffer         = NewSupvCommBuffer;
+  mMmUserCommonBufferPhysical = NewUserCommBuffer;
+  mMmSupvCommonBufferPhysical = NewSupvCommBuffer;
+
+  // Update supervisor communicate protocol communicate regions
+  mMmSupvCommunication.CommunicationRegion.PhysicalStart = (EFI_PHYSICAL_ADDRESS)(UINTN)NewSupvCommBuffer;
+  mMmSupvCommunication.CommunicationRegion.VirtualStart  = (EFI_PHYSICAL_ADDRESS)(UINTN)NewSupvCommBuffer;
+
+Done:
+  return Status;
+}
+
+/**
+  Entry Point for MM communication buffer driver DXE phase.
+
+  @param[in] ImageHandle  Image handle of this driver.
+  @param[in] SystemTable  A Pointer to the EFI System Table.
+
+  @retval EFI_SUCCESS
+  @return Others          Some error occurs.
+**/
+EFI_STATUS
+EFIAPI
+PublishMmCommunicationBuffer (
+  IN MM_SUPERVISOR_COMM_UPDATE_BUFFER  *UpdatedCommBuffer
+  )
+{
+  EFI_STATUS                               Status = EFI_NOT_FOUND;
+  UINT32                                   DescriptorSize;
+  EDKII_PI_SMM_COMMUNICATION_REGION_TABLE  *PiSmmCommunicationRegionTable;
+  EFI_MEMORY_DESCRIPTOR                    *Entry;
+  // MU_CHANGE Starts: MM_SUPV: Fetch allocated communication buffer from HOBs.
+  EFI_GUID                          *ConfTableGuid;
+  EFI_PEI_HOB_POINTERS              GuidHob;
+  MM_COMM_REGION_HOB                *CommRegionHob;
+  MM_SUPERVISOR_COMM_UPDATE_BUFFER  TempCommBuffer;
+  UINTN                             Index;
+
+  PiSmmCommunicationRegionTable = NULL;
+
+  if (UpdatedCommBuffer != NULL) {
+    CopyMem (&TempCommBuffer, UpdatedCommBuffer, sizeof (*UpdatedCommBuffer));
+  } else {
+    ZeroMem (&TempCommBuffer, sizeof (*UpdatedCommBuffer));

```

Comment:
I know this still results in `sizeof (MM_SUPERVISOR_COMM_UPDATE_BUFFER)` but it would be best to match the variable type being operated on here.

```suggestion
    ZeroMem (&TempCommBuffer, sizeof (TempCommBuffer));
```